src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/HashCode.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Interop
+{
+    public static class HashCode
+    {
+        public static int Combine(params object[] values)
+        {
+            int hash = 31;
+            foreach (object value in values)
+            {
+                hash = hash * 29 + value.GetHashCode();
```

Comment:
Where is this hash coming from?

Reply:
I think there is a well establish way of combining hashes of multiple objects. @stephentoub Do you know what I'm referring to here?

Reply:
System.HashCode

Reply:
Right. Let's get rid of this and use [`System.HashCode`](https://learn.microsoft.com/dotnet/api/system.hashcode).

Reply:
Hmmm. I wonder if there is a canonical way to do this. We have a few places in the generator that do this - see `CustomTypeMarshallers.GetHashCode()` and `ContainingSyntaxContext.GetHashCode()` for arbitrary collections. We also have a few for a specific numbers. I think I'd like to see a unified solution for this. Let's keep this for now, but update all places that compute a HashCode across multiple object instances with this implementation.

Reply:
We decided to use the `Hash.Combine(int, int)` from Roslyn.Utilities to avoid source build issues